pkg/virt-api/rest/authorizer_test.go

```diff
@@ -36,144 +36,162 @@ import (
 	"kubevirt.io/kubevirt/pkg/log"
 )
 
-var _ = Describe("VirtualMachineInstance Subresources", func() {
-	var server *ghttp.Server
-	var req *restful.Request
-
-	log.Log.SetIOWriter(GinkgoWriter)
-	fakecert := &x509.Certificate{}
-
-	app := authorizor{}
-	BeforeEach(func() {
-		req = &restful.Request{}
-		req.Request = &http.Request{}
-		req.Request.URL = &url.URL{}
-		req.Request.Header = make(map[string][]string)
-		req.Request.Header[userHeader] = []string{"user"}
-		req.Request.Header[groupHeader] = []string{"userGroup"}
-		req.Request.Header[userExtraHeaderPrefix+"test"] = []string{"userExtraValue"}
-		req.Request.URL.Path = "/apis/subresources.kubevirt.io/v1alpha3/namespaces/default/virtualmachineinstances/testvmi/console"
-
-		server = ghttp.NewServer()
-		config, err := clientcmd.BuildConfigFromFlags(server.URL(), "")
-		Expect(err).ToNot(HaveOccurred())
-
-		client, err := authorizationclient.NewForConfig(config)
-		Expect(err).ToNot(HaveOccurred())
-
-		app.subjectAccessReview = client.SubjectAccessReviews()
-		app.userHeaders = append(app.userHeaders, userHeader)
-		app.groupHeaders = append(app.groupHeaders, groupHeader)
-		app.userExtraHeaderPrefixes = append(app.userExtraHeaderPrefixes, userExtraHeaderPrefix)
-	})
-
-	Context("Subresource api", func() {
-		It("should reject unauthenticated user", func(done Done) {
-			allowed, reason, err := app.Authorize(req)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(allowed).To(Equal(false))
-			Expect(reason).To(Equal("request is not authenticated"))
-
-			close(done)
-		}, 5)
-
-		It("should reject unauthorized user", func(done Done) {
-
-			req.Request.TLS = &tls.ConnectionState{}
-			req.Request.TLS.PeerCertificates = append(req.Request.TLS.PeerCertificates, fakecert)
-
-			result, err := app.generateAccessReview(req)
+var _ = Describe("Authorizer", func() {
```

Comment:
note to reviewers: this looks worse (at least on github) than it is: I only put the complete old test in a new decsribe block, and added a new block at the end